src/private_cuda.h

```diff
@@ -128,4 +140,14 @@ struct _gpukernel {
 #endif
 };
 
+struct _gpucomm {
+  cuda_context* ctx;  // Start after the context
+// #ifdef WITH_CUDA_NCCL
+  ncclComm_t c;
+// #endif  // WITH_CUDA_NCCL
```

Comment:
Definition of `struct _gpucomm` should surely contain the ncclComm_t, but in the case that user does not fulfill the optional dep of nccl, the code should be able to be built nevertheless (that's why I will uncomment the defs around `ncclComm_t c` when code is overall cool). Right now I have these on comments, so I can work with autocompletion :stuck_out_tongue: . But I do not think we can somehow avoid the def around, because nccl is an optional dep.